### PR TITLE
DOCSUP-12413: macros support in functions cluster and clusterAllReplicas

### PR DESCRIPTION
**Pull request by olgarev**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

- Documentation for #26913

### CODE REVIEW

File: docs/ru/sql-reference/table-functions/cluster.md [hunk 1]
```diff
@@ -5,22 +5,44 @@ toc_title: cluster
...
+- `db.table` или `db`, `table` - имя базы данных и таблицы. 
+- `sharding_key` - ключ шардирования. Указывается, если данные добавляются более чем в один шард кластера. Необязательный аргумент.
+
+**Возвращаемое значение**
+
+Набор данных из кластеров.
+
+**Использование макросов**
+
+Аргументы могут содержать макросы — подстановки в фигурных скобках. Эти подстановки заменяются на соответствующие значения из конфигурационного файла, из секции [macros](../../operations/server-configuration-parameters/settings.md#macros).
```
Comment (olgarev) [inline docs/ru/sql-reference/table-functions/cluster.md:35]:
```suggestion
Аргумент `cluster` может содержать макрос — подстановку в фигурных скобках. Эта подстановка заменяется на соответствующее значение из конфигурационного файла, из секции [macros](../../operations/server-configuration-parameters/settings.md#macros).
```

File: docs/en/sql-reference/table-functions/cluster.md [hunk 2]
```diff
@@ -6,23 +6,41 @@ toc_title: cluster
...
+- `sharding_key` - When insert into cluster function with more than one shard, sharding key needs to be provided. Optional.
+
+**Returned value**
+
+The dataset from clusters.
 
-`cluster_name` – Name of a cluster that is used to build a set of addresses and connection parameters to remote and local servers.
+**Using Macros**
+
+Arguments can contain macros — substitutions in curly brackets. The substituted values are taken from the [macros](../../operations/server-configuration-parameters/settings.md#macros) section of the server configuration file.
```
Comment (olgarev) [inline docs/en/sql-reference/table-functions/cluster.md:35]:
```suggestion
`cluster_name` can contain macros — substitution in curly brackets. The substituted value is taken from the [macros](../../operations/server-configuration-parameters/settings.md#macros) section of the server configuration file.
```

File: docs/ru/sql-reference/table-functions/cluster.md [hunk 3]
```diff
@@ -5,22 +5,44 @@ toc_title: cluster
...
+
+```sql
+SELECT * FROM cluster('{cluster}', default.example_table);
 ```
 
-`cluster_name` – имя кластера, который обязан присутствовать в таблице `system.clusters`  и обозначает подмножество адресов и параметров подключения к удаленным и локальным серверам, входящим в кластер.
+**Использование и рекомендации** 
 
-Использование табличных функций `cluster` и `clusterAllReplicas` менее оптимальное чем создание таблицы типа `Distributed`, поскольку в этом случае соединение с сервером переустанавливается на каждый запрос. При обработке большого количества запросов, всегда создавайте `Distributed` таблицу заранее и не используйте табличные функции `cluster` и `clusterAllReplicas`.
+Использование табличных функций `cluster` и `clusterAllReplicas` менее оптимально, чем создание таблицы типа `Distributed`, поскольку в этом случае соединение с сервером переустанавливается на каждый запрос. При обработке большого количества запросов всегда создавайте `Distributed` таблицу заранее и не используйте табличные функции `cluster` и `clusterAllReplicas`.
```
Comment (adevyatova) [inline docs/ru/sql-reference/table-functions/cluster.md:45]:
```suggestion
Использование табличных функций `cluster` и `clusterAllReplicas` менее оптимально, чем создание таблицы типа `Distributed`, поскольку в этом случае при каждом новом запросе устанавливается новое соединение с сервером. При обработке большого количества запросов всегда создавайте `Distributed` таблицу заранее и не используйте табличные функции `cluster` и `clusterAllReplicas`.
```

File: docs/en/sql-reference/table-functions/cluster.md [hunk 4]
```diff
@@ -6,23 +6,41 @@ toc_title: cluster
...
 cluster('cluster_name', db.table[, sharding_key])
 cluster('cluster_name', db, table[, sharding_key])
 clusterAllReplicas('cluster_name', db.table[, sharding_key])
 clusterAllReplicas('cluster_name', db, table[, sharding_key])
 ```
+**Arguments**
+
+- `cluster_name` – Name of a cluster that is used to build a set of addresses and connection parameters to remote and local servers. 
+- `db.table` or `db`, `table` - Name of a database and a table.  
+- `sharding_key` - When insert into cluster function with more than one shard, sharding key needs to be provided. Optional.
```
Comment (adevyatova) [inline docs/en/sql-reference/table-functions/cluster.md:27]:
```suggestion
- `sharding_key` -  A sharding key if the cluster has more than one shard. Optional.
```

File: docs/en/sql-reference/table-functions/cluster.md [hunk 5]
```diff
@@ -6,23 +6,41 @@ toc_title: cluster
...
 cluster('cluster_name', db.table[, sharding_key])
 cluster('cluster_name', db, table[, sharding_key])
 clusterAllReplicas('cluster_name', db.table[, sharding_key])
 clusterAllReplicas('cluster_name', db, table[, sharding_key])
 ```
+**Arguments**
+
+- `cluster_name` – Name of a cluster that is used to build a set of addresses and connection parameters to remote and local servers. 
+- `db.table` or `db`, `table` - Name of a database and a table.  
+- `sharding_key` -  A sharding key if the cluster has more than one shard. Optional.
```
Comment (olgarev) [inline docs/en/sql-reference/table-functions/cluster.md:27]:
```suggestion
- `sharding_key` -  A sharding key. Optional. Needs to be specified if the cluster has more than one shard. 
```

File: docs/ru/sql-reference/table-functions/cluster.md [hunk 6]
```diff
@@ -5,22 +5,44 @@ toc_title: cluster
...
+cluster('cluster_name', db.table[, sharding_key])
+cluster('cluster_name', db, table[, sharding_key])
+clusterAllReplicas('cluster_name', db.table[, sharding_key])
+clusterAllReplicas('cluster_name', db, table[, sharding_key])
+```
+**Аргументы**
+
+- `cluster_name` – имя кластера, который обозначает подмножество адресов и параметров подключения к удаленным и локальным серверам, входящим в кластер.
+- `db.table` или `db`, `table` - имя базы данных и таблицы. 
+- `sharding_key` - ключ шардирования. Указывается, если данные добавляются более чем в один шард кластера. Необязательный аргумент.
```
Comment (olgarev) [inline docs/ru/sql-reference/table-functions/cluster.md:27]:
```suggestion
- `sharding_key` - ключ шардирования. Необязательный аргумент. Указывается, если данные добавляются более чем в один шард кластера. 
```

**Comment by olgarev** [comment]
@tavplubix Hi, please would you review this PR?

**Review by tavplubix — APPROVED** [review]

**Pull request merged**